sdk/core/azure-core/src/http/response.cpp

```diff
@@ -16,35 +18,42 @@ std::string const& Response::GetReasonPhrase() { return m_reasonPhrase; }
 
 std::map<std::string, std::string> const& Response::GetHeaders() const { return this->m_headers; }
 
-void Response::AddHeader(std::string const& header)
+void Response::AddHeader(uint8_t const* const begin, uint8_t const* const last)
 {
   // get name and value from header
-  auto start = header.begin();
-  auto end = std::find(start, header.end(), ':');
+  auto start = begin;
+  auto end = std::find(start, last, ':');
 
-  if (end == header.end())
+  if (end == last)
   {
     return; // not a valid header or end of headers symbol reached
   }
 
+  // Always toLower() headers
+  // auto headerName = Azure::Core::Details::ToLower(std::string(start, end));
```

Comment:
Remove this commented code

Reply:
But we do want header keys lowercased

Reply:
yeah, I commented it for now after talking to @JinmingHu-MSFT .
Since this is a breaking change, I will enable header names lowercasing in a separate PR so storage team can make the required changes for Storage code